src/components/dock/default-dock-button/default-dock-button.tsx

```diff
@@ -0,0 +1,142 @@
+import { Component, Event, EventEmitter, h, Prop, State } from '@stencil/core';
+import { CustomDockButton, DockItem } from '../dock.types';
+import { createRandomString } from '../../../util/random-string';
+
+/**
+ * @private
+ */
+@Component({
+    tag: 'limel-default-dock-button',
+    shadow: false,
+    styleUrl: 'default-dock-button.scss',
+})
+export class DefaultDockButton implements CustomDockButton {
```

Comment:
It may look odd that `limel-default-dock-button` implements `CustomDockButton`, but if you see it from the point of view of the consumer, `CustomDockButton` is the interface you should implement if you wish to make a custom dock button!